pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -83,8 +111,33 @@ func CreateCloudProvider(vendorVersion string) (*CloudProvider, error) {
 
 }
 
-func createBetaCloudService(vendorVersion string) (*beta.Service, error) {
-	client, err := newDefaultOauthClient()
+func generateTokenSource(configFile *ConfigFile) (tokenSource oauth2.TokenSource, err error) {
+	if configFile != nil {
+		if configFile.Global.TokenURL != "" {
+			// if tokenURL is nil, set tokenSource to nil. This will force the OAuth client to fall
+			// back to use DefaultTokenSource. This allows running gceCloud remotely.
+			if configFile.Global.TokenURL == "nil" {
```

Comment:
yikes, do you actually see string "nil" sometimes?

Reply:
I haven't seen it myself. This was copied from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/gce/gce.go#L293

What would you recommend?